core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java

```diff
@@ -0,0 +1,237 @@
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import alluxio.RuntimeConstants;
+import alluxio.master.block.BlockMaster;
+import alluxio.master.file.FileSystemMaster;
+import alluxio.master.journal.Journal;
+import alluxio.metrics.MetricsSystem;
+import alluxio.wire.WorkerInfo;
+import alluxio.web.MasterUIWebServer;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.util.List;
+import java.util.SortedMap;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+import javax.servlet.ServletContext;
+
+/**
+ * Unit tests for {@link AlluxioMasterRestServiceHandler}.
+ */
+
+public class AlluxioMasterRestServiceHandlerTest {
+  private AlluxioMaster mMaster;
+  private ServletContext mContext;
+  private BlockMaster mBlockMaster;
+  private AlluxioMasterRestServiceHandler mHandler;
+
+  @BeforeClass
+  public static void setUpClass() {
+    String filesPinnedProperty =
+        MetricsSystem.getMasterMetricName(FileSystemMaster.Metrics.FILES_PINNED);
+    MetricsSystem.METRIC_REGISTRY.remove(filesPinnedProperty);
+  }
+
+  @Before
+  public void setUp() {
+    mMaster = mock(AlluxioMaster.class);
+    mContext = mock(ServletContext.class);
+    Journal journal = mock(Journal.class);
+    mBlockMaster = new BlockMaster(journal);
+    when(mMaster.getBlockMaster()).thenReturn(mBlockMaster);
+    when(mContext.getAttribute(MasterUIWebServer.ALLUXIO_MASTER_SERVLET_RESOURCE_KEY))
+        .thenReturn(mMaster);
+    when(mMaster.getMasterAddress()).thenReturn(new InetSocketAddress("localhost", 8080));
+    when(mContext.getAttribute(MasterUIWebServer.ALLUXIO_MASTER_SERVLET_RESOURCE_KEY))
+        .thenReturn(mMaster);
+    mHandler = new AlluxioMasterRestServiceHandler(mContext);
+  }
+
+  @Test
+  public void testGetConfiguration() {
+    Response response = mHandler.getConfiguration();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertTrue("Entry must be a SortedMap!", (response.getEntity() instanceof SortedMap));
+    SortedMap<String, String> entry = (SortedMap<String, String>) response.getEntity();
+    assertFalse("Properties Map must be not empty!", (entry.isEmpty()));
+  }
+
+  @Test
+  public void testGetRpcAddress() {
+    Response response = mHandler.getRpcAddress();
+    assertNotNull("Response must be not null!", response);
+    assertNotNull("Response must have a entry!", response.getEntity());
+    assertEquals("Entry must be a String!", String.class, response.getEntity().getClass());
```

Comment:
we should consistently use either instanceof or check the equality of the classes, whichever approach you prefer.